airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,159 @@
+
+import pytest
+import responses
+from airbyte_cdk.models import SyncMode
+from jsonschema import validate
+from source_amazon_ads import SourceAmazonAds
+
+
+def setup_responses(profiles_response=None, campaigns_response=None):
+    responses.add(
+        responses.POST,
+        "https://api.amazon.com/auth/o2/token",
+        json={"access_token": "alala", "expires_in": 10},
+    )
+    if profiles_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/v2/profiles",
+            body=profiles_response,
+        )
+    if campaigns_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/campaigns",
+            body=campaigns_response,
+        )
+
+
+@responses.activate
+def test_streams_profile(test_config, profiles_response):
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    assert len(streams) == 6
+    profile_stream = streams[0]
+    assert profile_stream.name == "profiles"
+    schema = profile_stream.get_json_schema()
+    setup_responses(profiles_response=profiles_response)
+
+    records = profile_stream.read_records(SyncMode.full_refresh)
+    records = [r for r in records]
+    assert len(responses.calls) == 2
+    assert len(profile_stream.ctx.profiles) == 4
+    assert len(records) == 4
+    expected_records = loads(profiles_response)
+    for record, expected_record in zip(records, expected_records):
+        validate(schema, record)
+        assert record == expected_record
+
+
+@responses.activate
+def test_streams_campaigns_one_vendor(test_config, profiles_response, campaigns_response):
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = streams[0]
+    assert profile_stream.name == "profiles"
+    campaigns_stream = streams[1]
+    assert campaigns_stream.name == "sponsored_display_campaigns"
+    setup_responses(profiles_response=profiles_response)
+
+    responses.add(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        body=campaigns_response,
+    )
+    records = profile_stream.read_records(SyncMode.full_refresh)
+    _ = [r for r in records]
+
+    records = campaigns_stream.read_records(SyncMode.full_refresh)
+    campaigns_records = [r for r in records]
+    assert len(campaigns_records) == 4  # we have only one vendor account
+
+
+@responses.activate
+def test_streams_campaigns_4_vendors(test_config, profiles_response, campaigns_response):
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = streams[0]
+    assert profile_stream.name == "profiles"
+    campaigns_stream = streams[1]
+    profiles_response = loads(profiles_response)
+    for profile in profiles_response:
+        profile["accountInfo"]["type"] = "vendor"
+    profiles_response = dumps(profiles_response)
+    assert campaigns_stream.name == "sponsored_display_campaigns"
```

Comment:
Why don't you move this assertion below line 105? So the code will look like this:
```
campaigns_stream = streams[1]
assert campaigns_stream.name == "sponsored_display_campaigns"
```

Or move `campaigns_stream = streams[1]` above line 110.